open-apparel-registry/open-apparel-registry
# Integrate facility/processing type taxonomy into facility claims

## Overview

Makes various changes to the facility claims form and back-end to integrate the new facility/processing type taxonomy.

- Update the extended field creation for claims to create fields for processing type, facility type, and parent company. The logic of determining the values for these fields was refactored out so that it could be reused for claims and facilities.
- Choice tuples are generated for the processing and facility types so that they can be used on the claims form and for the claims model. A test was updated to accommodate these changes.
- FacilityClaims.FacilityType is lengthened to support multiple values in the facility type taxonomy.
- Change "production type" to "processing type" on the claims form. "Production type" is the old name for "processing type". No variables names were changed at this time for the sake of simplicity.
- FacilityClaims.FacilityType was pluralized on the front-end since users can now select multiple values for this field.

Connects #1590

## Demo

![Feb-11-2022 11-09-21](https://user-images.githubusercontent.com/1042475/153627056-da433159-7334-4453-a722-5d4d6e324cfb.gif)

## Notes

Follow-up work to be done:

- Generate data migrations for facility claims on production that have a processing and/or facility type already to move them to a value within the taxonomy. A special case to be aware of is that "other" type for facilities. 
- Consider renaming FacilityClaim.ProductionType to ProcessingType, either just related variables, and/or the back-end model field. This will require a migration. 

## Testing Instructions

- Run `./scripts/update` to apply the included migration.
- Create or a visit an existing facility claim. Example: http://localhost:6543/claimed/1/
- Edit/add facility type, processing type, and parent company and save the changes.
- Visit the facility on the main map, and verify the added/edited values for the extended fields are displayed.
- Make edits to the initial values you set (for example, remove some processing types), and verify those edits are displayed on the main map.

## Checklist
